File: src/main/java/org/nia/logic/lists/Location.java
package org.nia.logic.lists;

import org.nia.logic.quests.QuestsEnum;

import java.util.Arrays;
import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;

/**
 * @author IANazarov
 */
public enum Location {
    TAVERN(),
    QUEST();


    public static QuestsEnum getRandomQuest() {
        List<QuestsEnum> list = Arrays.stream(QuestsEnum.values()).collect(Collectors.toList());
        return list.get(new Random().nextInt(list.size()));
    }
}

File: src/main/java/ru/nia/tavern/model/types/Location.kt
package ru.nia.tavern.model.types

import ru.nia.tavern.quests.QuestsEnum
import java.util.*
import java.util.stream.Collectors

/**
 * @author IANazarov
 */
enum class Location {
    TAVERN,
    QUEST;


    companion object {


        val randomQuest: QuestsEnum
            get() {
                val list = Arrays.stream(QuestsEnum.values()).filter(Predicate<QuestsEnum> { it.isRunnable() }).collect<List<QuestsEnum>, Any>(Collectors.toList())
                return list[Random().nextInt(list.size)]
            }
    }
}
